explorer/src/pages/Module/components/ModulePortals/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { t } from "i18next";
+import { useRef } from "react";
+import useSWR from "swr";
+import { Address } from "viem";
+
+import { DataTable } from "@/components/DataTable";
+import { columns, portalColumnsOption, skeletonPortals } from "@/constants/columns/portal";
+import { columnsSkeleton } from "@/constants/columns/skeleton";
+import { SWRKeys } from "@/interfaces/swr/enum";
+import { useNetworkContext } from "@/providers/network-provider/context";
+import { APP_ROUTES } from "@/routes/constants";
+import { isNotNullOrUndefined } from "@/utils";
+
+export const ModulePortals: React.FC<{ moduleId: Address }> = ({ moduleId }) => {
+  const {
+    sdk,
+    network: { chain },
+  } = useNetworkContext();
+
+  const { data: portals, isLoading } = useSWR(
+    `${SWRKeys.GET_MODULE_PORTAL_LIST}/${moduleId}/${chain.id}`,
+    async () => {
+      const allPortals = await sdk.portal.findBy(100, 0);
+      const portalList = allPortals.filter((portal) => portal.modules.includes(moduleId));
+      return portalList.slice(0, 5).map((portal) => ({
+        ...portal,
+        id: portal.id as `0x${string}`,
+      }));
```

Comment:
This can be simplified a lot 🙁 

You can leverage something like `sdk.portal.findBy(5, 0, { modules_contains: ["MODULE_ID"] }, "attestationCounter", "desc")`
=> This fetches 5 Portals at the most, where the `modules` array contains the given Module ID, and orders the results starting with the Portal that issued the most Attestations.

You can test your requests to the subgraph to optimize them by directly going to the subgraph's URL ([here](https://api.studio.thegraph.com/query/67521/verax-v2-linea/v0.0.1/graphql?) for Linea mainnet for exampel, all the links are available [here](https://github.com/Consensys/linea-attestation-registry?tab=readme-ov-file#subgraph-v2)).

Reply:
done i think